lib/puppet/util/windows/service.rb

```diff
@@ -0,0 +1,708 @@
+require 'puppet/util/windows'
+require 'ffi'
+
+module Puppet::Util::Windows
+  # This module is designed to provide an API between the windows system and puppet for
+  # service management.
+  #
+  # for an overview of the service state transitions see: https://docs.microsoft.com/en-us/windows/desktop/Services/service-status-transitions
+  module Service
+    extend FFI::Library
+    extend Puppet::Util::Windows::String
+
+    # Service control codes
+    # https://docs.microsoft.com/en-us/windows/desktop/api/Winsvc/nf-winsvc-controlservice
+    SERVICE_CONTROL_CONTINUE       = 0x00000003
```

Comment:
I'm sure everything here for services is probably fine, but you might want to double check the constants against the latest copy of the Windows 10 SDK. We found values missing from MSDN, but that were present in the actual headers during the scheduled task rewrite.

In particular some of the latter version constants in https://github.com/puppetlabs/puppetlabs-scheduled_task/blob/master/lib/puppet_x/puppetlabs/scheduled_task/task.rb#L29-L41 were simply missing from online docs.